endpoints/openrtb2/auction_test.go

```diff
@@ -6108,3 +6266,654 @@ func TestValidateAliases(t *testing.T) {
+											{
+												Bid:          &openrtb2.Bid{ImpID: "imp1"},
+												NonBidReason: 100,
+											},
+										},
+									}),
+								},
+							},
+						},
+					},
+				},
+				{
+					Stage: hooks.StageBidderRequest.String(),
+					Groups: []hookexecution.GroupOutcome{
+						{
+							InvocationResults: []hookexecution.HookOutcome{
+								{
+									Status: hookexecution.StatusSuccess,
+									SeatNonBid: getNonBids(map[string][]openrtb_ext.NonBidParams{
+										"pubmatic": {
+											{
+												Bid:          &openrtb2.Bid{ImpID: "imp2"},
+												NonBidReason: 100,
+											},
+										},
+									}),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNonBids: getNonBids(map[string][]openrtb_ext.NonBidParams{
+				"pubmatic": {
+					{
+						Bid:          &openrtb2.Bid{ImpID: "imp1"},
+						NonBidReason: 100,
+					},
+					{
+						Bid:          &openrtb2.Bid{ImpID: "imp2"},
+						NonBidReason: 100,
+					},
+				},
+			}),
+		},
+		{
+			name: "multi group outcomes with empty nonbids",
+			stageOutcomes: []hookexecution.StageOutcome{
+				{
+					Stage: hooks.StageAllProcessedBidResponses.String(),
+					Groups: []hookexecution.GroupOutcome{
+						{
+							InvocationResults: []hookexecution.HookOutcome{
+								{
+									Status:     hookexecution.StatusSuccess,
+									SeatNonBid: openrtb_ext.NonBidCollection{},
+								},
+							},
+						},
+						{
+							InvocationResults: []hookexecution.HookOutcome{
+								{
+									Status:     hookexecution.StatusSuccess,
+									SeatNonBid: openrtb_ext.NonBidCollection{},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNonBids: openrtb_ext.NonBidCollection{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nonBids := getNonBidsFromStageOutcomes(tt.stageOutcomes)
+			assert.Equal(t, nonBids, tt.expectedNonBids, "getNonBidsFromStageOutcomes returned incorrect nonBids")
+		})
+	}
+}
+
+// getNonBids is utility function which forms NonBidCollection from NonBidParams input
+func getNonBids(bidParamsMap map[string][]openrtb_ext.NonBidParams) openrtb_ext.NonBidCollection {
+	nonBids := openrtb_ext.NonBidCollection{}
+	for bidder, bidParams := range bidParamsMap {
+		for _, bidParam := range bidParams {
+			nonBid := openrtb_ext.NewNonBid(bidParam)
+			nonBids.AddBid(nonBid, bidder)
+		}
+	}
+	return nonBids
+}
+
+func TestSeatNonBidInAuction(t *testing.T) {
+	type args struct {
+		bidRequest                openrtb2.BidRequest
+		seatNonBidFromHoldAuction openrtb_ext.NonBidCollection
+		errorFromHoldAuction      error
+		rejectRawAuctionHook      bool
+		errorFromHook             error
+	}
+	type want struct {
+		statusCode int
+		body       string
+		seatNonBid []openrtb_ext.SeatNonBid
+	}
+	testCases := []struct {
+		description string
+		args        args
+		want        want
+	}{
+		{
+			description: "request parsing failed, auctionObject should contain seatNonBid",
```

Comment:
This part is not clear to me. `parseRequest` has nothing to do with bid responses. Do we still want to include `seatNonBid` to the auction response?

Reply:
As part of the SeatNonBid proposal (https://github.com/prebid/prebid-server/issues/2367) , seatNonBid is not just  related to bid-responses. 
Please refer NBR codes (202: Error - bad input parameters, 116: No Bid - Incomplete SupplyChain) this needs to be capture inside `parseRequest`.